ibis/expr/types/temporal.py

```diff
@@ -19,9 +49,237 @@ class TemporalColumn(AnyColumn, TemporalValue):
     pass  # noqa: E701,E302
 
 
+class _DateComponentMixin(TemporalValue):
```

Comment:
Can we erase the parent class here and explicitly inherit from that in the concrete classes keeping the mixins as pure interfaces?

Reply:
Yep, good idea.